### ISSUE TITLE
Feature request: type based autocompletion

### ISSUE THREAD

**Issue by dlesbre** [ocaml/ocaml-lsp#807]
**Disclaimer:** I'm not sure if this is the right place to post this. It might need to go to [ocamllabs/vscode-ocaml-platform](https://github.com/ocamllabs/vscode-ocaml-platform).

One feature I'd really like is some sort of type base autocompletion for incomplete matches and modules. For example a code action or quick fix that could take this code:
```ocaml
type t = Int of int | Bool of bool | Char of char

let foo = function
  | Int i -> Int (i+1)
```
and autocomplete it to something like:
```ocaml
type t = Int of int | Bool of bool | Char of char

let foo function
  | Int i -> Int (i+1)
  | Bool _ -> _
  | Char _ -> _
```
Or similarly with modules:
```ocaml
module type A = sig 
  val foo : int
  val bar : bool
end

module S : A = struct end

module F(X : A) = ...

module G = F(struct 
  let foo = 2
end)
```
And complete it into:
```ocaml
module type A = sig 
  val foo : int
  val bar : bool
end

module S : A = struct let foo = _ let bar = _ end

module F(X : A) = ...

module G = F(struct 
  let foo = 2
  let bar = _
end)
```
I think it should be doable since in both cases we get an error message or warning mentioning non-exhaustive pattern matching with examples of missing cases and missing values.

**Comment by ulugbekna** [comment]
I think we have what you're asking? See: 

https://user-images.githubusercontent.com/16353531/181725692-70259a25-0eb4-4c97-9c3d-fa563979b8c4.mov

https://user-images.githubusercontent.com/16353531/181725804-040752a9-726c-42c3-8d9a-651705e5b00f.mov

We should make these things more discoverable.

**Comment by dlesbre** [comment]
You're right, I found the destruct (`Alt-D` on VSCode on Linux) for the match. I don't seem to get the module autocompletion though, but maybe that is just a VSCode issue and not an LSP one.

**Comment by ulugbekna** [comment]
Documented `destruct` and `construct` functionality in the README.
